osbs/core.py

```diff
@@ -382,6 +382,45 @@ def wait_for_new_build_config_instance(self, build_config_id, prev_version):
 
         raise OsbsResponseException("New BuildConfig instance not found")
 
+    def stream_logs(self, build_id):
+        """
+        stream logs from build
+
+        :param build_id: str
+        :return: iterator
+        """
+        kwargs = {'follow': 1}
+
+        # If connection is closed within this many seconds, give up:
+        min_idle_timeout = 60
+
+        while True:
+            buildlogs_url = self._build_url("builds/%s/log/" % build_id,
+                                            **kwargs)
+            response = self._get(buildlogs_url, stream=1,
+                                 headers={'Connection': 'close'})
+            check_response(response)
+
+            # Stream logs, but be careful of the connection closing
+            # due to idle timeout. In that case, try again until the
+            # call returns more quickly than a reasonable timeout
+            # would be set to.
+            last_activity = time.time()
+
+            for line in response.iter_lines():
+                last_activity = time.time()
+                yield line
+
+            idle = time.time() - last_activity
+            logger.debug("connection closed after %ds", idle)
+            if idle < min_idle_timeout:
```

Comment:
We could. However: it would slow down most builds slightly because when the build finishes we'd have an extra server call to fetch build JSON to check it's really finished.

(FWIW, koji-containerbuild already does this call)